src/keystore.c

```diff
@@ -427,6 +428,20 @@ static bool _get_xprv(const uint32_t* keypath, const size_t keypath_len, struct
     return true;
 }
 
+bool keystore_get_root_fingerprint(uint8_t* fingerprint)
+{
+    struct ext_key derived_xpub __attribute__((__cleanup__(keystore_zero_xkey))) = {0};
+    if (!keystore_get_xpub(NULL, 0, &derived_xpub)) {
+        return false;
+    }
+    unsigned char fingerprint_buffer[4];
```

Comment:
i think you can drop this variable and the memcpy below, and just pass `fingerprint` to `bip32_key_get_fingerprint`.

If you want to keep the buffer size safety check, in commander.c, you could call `keystore_get_root_fingerprint(response->fingerprint, sizeof(response->fingerprint))` and pass the size to `bip32_key_get_fingerprint()`, which will make sure it is 4 bytes.

Reply:
Removed the extra buffer, I won't add an explicit size check.